pkg/yurthub/cachemanager/cache_manager_test.go

```diff
@@ -174,20 +181,81 @@ func TestCacheResponse(t *testing.T) {
 			accept:     "application/json",
 			verb:       "GET",
 			path:       "/api/v1/nodes/mynode2",
+			resource:   "nodes",
 			namespaced: false,
 			expectResult: expectData{
 				rv:   "6",
 				name: "mynode2",
 				kind: "Node",
 			},
 		},
+
+		//used to test whether custom resources can be cached correctly
+		{
+			desc:    "cache response for get crontab1",
+			group:   "stable.example.com",
+			version: "v1",
+			key:     "kubelet/crontabs/default/crontab1",
+			inputObj: runtime.Object(&unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "stable.example.com/v1",
+					"kind":       "CronTab",
+					"metadata": map[string]interface{}{
+						"name":            "crontab1",
+						"namespace":       "default",
+						"resourceVersion": "1",
+					},
+				},
+			}),
+			userAgent:  "kubelet",
+			accept:     "application/json",
+			verb:       "GET",
+			path:       "/apis/stable.example.com/v1/namespaces/default/crontabs/crontab1",
+			resource:   "crontabs",
+			namespaced: true,
+			expectResult: expectData{
+				rv:   "1",
+				name: "crontab1",
+				ns:   "default",
+				kind: "CronTab",
+			},
+		},
+		{
+			desc:    "cache response for get crontab2",
+			group:   "stable.example.com",
+			version: "v1",
+			key:     "kubelet/crontabs/default/crontab2",
+			inputObj: runtime.Object(&unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "stable.example.com/v1",
+					"kind":       "CronTab",
+					"metadata": map[string]interface{}{
+						"name":            "crontab2",
+						"namespace":       "default",
+						"resourceVersion": "3",
+					},
+				},
+			}),
+			userAgent:  "kubelet",
+			accept:     "application/json",
+			verb:       "GET",
+			path:       "/apis/stable.example.com/v1/namespaces/default/crontabs/crontab2",
+			resource:   "crontabs",
+			namespaced: true,
+			expectResult: expectData{
+				rv:   "3",
+				name: "crontab2",
+				ns:   "default",
+				kind: "CronTab",
+			},
+		},
```

Comment:
how about add a namespaced=false test case?

Reply:
Test cases is already added, thanks for review.